EpiResponsivePicture/TagBuilders/PictureTagBuilder.cs

```diff
@@ -0,0 +1,154 @@
+using EPiServer;
+using EPiServer.Core;
+using EPiServer.ServiceLocation;
+using EPiServer.Web.Routing;
+using Forte.EpiResponsivePicture.ResizedImage;
+using Forte.EpiResponsivePicture.ResizedImage.Property;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.Toolkit.Diagnostics;
+
+namespace Forte.EpiResponsivePicture.TagBuilders
+{
+    public class PictureTagBuilder : IPictureTagBuilder
+    {
+        private readonly TagBuilder element;
+        private PictureProfile pictureProfile;
+        private ContentReference pictureContentReference;
+        private string pictureFallbackUrl;
+        private ResizedPictureViewModel pictureViewModel;
+        private FocalPoint focalPoint;
+        private string pictureUrl;
+        private string altImgText;
+        
+        private PictureTagBuilder()
+        {
+            element = new TagBuilder("picture");
+        }
+
+        public static IPictureTagBuilder Create() => new PictureTagBuilder();
+
+        public IPictureTagBuilder WithProfile(PictureProfile profile)
+        {
+            pictureProfile = profile;
+            return this;
+        }
+
+        public IPictureTagBuilder WithContentReference(ContentReference contentReference)
+        {
+            pictureContentReference = contentReference;
+            return this;
+        }
+
+        public IPictureTagBuilder WithFallbackUrl(string fallbackUrl)
+        {
+            pictureFallbackUrl = fallbackUrl;
+            return this;
+        }
+
+        public IPictureTagBuilder WithViewModel(ResizedPictureViewModel viewModel)
+        {
+            pictureViewModel = viewModel;
+            return this;
+        }
+
+        public TagBuilder Build()
+        {
+            Guard.IsNotNull(pictureProfile, nameof(pictureProfile));
+            
+            pictureFallbackUrl ??= string.Empty;
+            pictureViewModel ??= new ResizedPictureViewModel();
+            
+            GetPictureData();
+
+            var sourceTagBuilder = SourceTagBuilder
+                .Create()
+                .WithImageUrl(pictureUrl)
+                .WithFocalPoint(focalPoint)
+                .WithResizedImageFormat(pictureProfile.Format);
+
+            foreach (var pictureSource in pictureProfile.Sources)
+            {
+                var sourceTag = sourceTagBuilder.WithSource(pictureSource).Build();
+
+                element.InnerHtml.AppendHtml(sourceTag.RenderSelfClosingTag());
+                
+                sourceTagBuilder.Clear();
+            }
+            
+            AddAdditionalAttributes();
+
+            var imgTag = CreateImageElement();
+            element.InnerHtml.AppendHtml(imgTag.RenderSelfClosingTag());
+            
+            return element;
+        }
+        
+        private void GetPictureData()
```

Comment:
IMHO all these Get* methods look odd when they return nothing (void). These seem to be rather Set than Get, don't you agree?